### PR TITLE
Update tests to remove additional AWS auth env

### PR DESCRIPTION
**Pull request by atward**

### CODE REVIEW

File: tests/amazon/test_credentials.py [hunk 1]
```diff
@@ -31,10 +31,11 @@
...
 
-        # Moto provides the following credentials from 169.254.269.254
-        self.assertEquals(aws_creds.access_key, 'test-key')
-        self.assertEquals(aws_creds.secret_key, 'test-secret-key')
-        self.assertEquals(aws_creds.token, 'test-session-token')
+        # Moto provides 'test-*' via instance_metadata mock
+        # Moto provides '*EXAMPLE*' via sts mock
+        self.assertTrue(aws_creds.access_key, ['test-key', 'AKIAIOSFODNN7EXAMPLE'])
+        self.assertTrue(aws_creds.secret_key in ['test-secret-key', 'aJalrXUtnFEMI/K7MDENG/bPxRfiCYzEXAMPLEKEY'])
+        self.assertTrue(aws_creds.token in ['test-session-token', 'BQoEXAMPLEH4aoAH0gNCAPyJxz4BlCFFxWNE1OPTgk5TthT+FvwqnKwRcOIfrRh3c/LTo6UDdyJwOOvEVPvLXCrrrUtdnniCEXAMPLE/IvU1dYUg2RVAJBanLiHb4IgRmpRV3zrkuWJOgQs8IZZaIv2BXIa2R4OlgkBN9bkUDNCJiBeb/AXlzBBko7b15fjrBs2+cTQtpZ3CYWFXG8C5zqx37wnOE49mRl/+OtkIKGO7fAE'])
```
Comment (delfick) [inline tests/amazon/test_credentials.py]: but it shouldn't be from sts at all in this test because we aren't assuming a role.

Comment (atward) [reply tests/amazon/test_credentials.py]: So the test should read `if not instance_metadata: skiptest`?

This test currently fails when not running in EC2.

Comment (delfick) [reply tests/amazon/test_credentials.py]: nah, moto fakes a 169.254.169.254

Comment (delfick) [reply tests/amazon/test_credentials.py]: that's where the test-* results come from

Comment (atward) [reply tests/amazon/test_credentials.py]: Yep, see that. There were some vars missing from the os.environ purge which were/could cause issues.

**Comment by delfick** [comment]
lgtm

**Pull request merged**